src/Facets/HopFacetPacked.sol

```diff
@@ -6,12 +6,40 @@ import { ERC20 } from "solmate/utils/SafeTransferLib.sol";
 import { LibAsset, IERC20 } from "../Libraries/LibAsset.sol";
 import { TransferrableOwnership } from "../Helpers/TransferrableOwnership.sol";
 import { HopFacetOptimized } from "lifi/Facets/HopFacetOptimized.sol";
+import { WETH } from "solmate/tokens/WETH.sol";
+
+interface L2_AmmWrapper {
+    function bridge() external view returns (address);
+
+    function l2CanonicalToken() external view returns (address);
+
+    function hToken() external view returns (address);
+
+    function exchangeAddress() external view returns (address);
+}
+
+interface Swap {
+    function swap(
+        uint8 tokenIndexFrom,
+        uint8 tokenIndexTo,
+        uint256 dx,
+        uint256 minDy,
+        uint256 deadline
+    ) external returns (uint256);
+}
 
 /// @title Hop Facet (Optimized for Rollups)
 /// @author LI.FI (https://li.fi)
 /// @notice Provides functionality for bridging through Hop
 /// @custom:version 1.0.2
 contract HopFacetPacked is ILiFi, TransferrableOwnership {
+    /// Storage ///
+
+    address public immutable bridge;
+    address public immutable l2CanonicalToken;
+    address public immutable hToken;
+    address public immutable exchangeAddress;
```

Comment:
I would rename the variables to highlight that they are for native token only. ERC20 tokens use different addresses.